ecosystem/node-checker/src/evaluator/traits.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{configuration::EvaluatorArgs, evaluator::EvaluationResult};
+use std::{error::Error, fmt::Debug};
+
+// You'll notice that a couple of the methods here require Self: Sized.
+// The intent with the Evaluator trait is that a caller will build a vec
+// of different evaluators with different associated input and error types.
+// To do this, we need to construct trait objects (Box<dyn Evaluator<Input, Error>>),
+// and to do that, the trait needs to be object safe. Object safe traits may
+// not have functions or methods that do not take self as a parameter unless
+// they are marked as `where Self: Sized`. This constraint means these functions
+// can not be called on the trait object itself, just on the struct / Self.
+// For more information on this topic, see:
+// https://doc.rust-lang.org/reference/items/traits.html#object-safety
+
+/// An Evaluator is a component of NHC that is responsible for evaluating
+/// a particular aspect of the node under investigation, be that metrics,
+/// system information, API checks, load tests, etc.
+#[async_trait::async_trait]
```

Comment:
Silly question: where did the async requirement come from?

Reply:
For future evaluators where they need to do something async, e.g. the TPS evaluator, latency evaluator, etc. Up until now the evaluators had what they needed passed in, but other evaluators don't necessarily work like that.